source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc

```diff
@@ -143,12 +146,19 @@ SimpleHttpCache::varyLookup(const LookupRequest& request,
   // This method should be called from lookup, which holds the mutex for reading.
   mutex_.AssertReaderHeld();
 
-  const auto vary_header = response_headers->get(Http::CustomHeaders::get().Vary);
-  ASSERT(!vary_header.empty());
+  absl::btree_set<absl::string_view> vary_header_values =
+      VaryHeader::getVaryValues(*response_headers);
+  ASSERT(!vary_header_values.empty());
 
   Key varied_request_key = request.key();
-  const std::string vary_key = VaryHeader::createVaryKey(vary_header, request.getVaryHeaders());
-  varied_request_key.add_custom_fields(vary_key);
+  const absl::optional<std::string> vary_identifier =
+      request.vary_allow_list().createVaryIdentifier(vary_header_values, request.request_headers());
+  if (!vary_identifier.has_value()) {
+    // The vary allow list has changed and has made the vary header of this
```

Comment:
Doesn't vary allow list come from the CacheConfig only? Should that change the listener along with the filter-chain will get re-created. Could you explain what you had in mind here?

Reply:
This would happen if an entry gets cached, then the config changes and a subsequent request for the same content hits that cache entry, but now the value the entry varied on is no longer permitted. I defaulted to treating it as a cache miss so we'd reinsert from upstream (or choose not to cache, should upstream still provide a vary header value that is not allowed).

Reply:
I didn't realise `HttpCacheFactory` returns the same instance of `SimpleHttpCache` when `getCache()` is called. Makes sense and please ignore my comment.